service/history/historyEngine.go

```diff
@@ -436,6 +454,30 @@ func (e *historyEngineImpl) registerDomainFailoverCallback() {
 				e.timerProcessor.NotifyNewTimers(e.currentClusterName, fakeDecisionTimeoutTask)
 			}
 
+			// handle graceful failover on active to passive
+			// make sure task processor failover the domain before inserting the failover marker
+			failoverMarkerTasks := []*persistence.FailoverMarkerTask{}
+			for _, nextDomain := range nextDomains {
+				failoverMarkerInsertion(shardNotificationVersion, nextDomain, func() {
```

Comment:
Do we still need this function? seems like the checks can be done inline.

Reply:
Updated